python/lsst/sims/featureScheduler/surveys.py

```diff
@@ -785,6 +804,656 @@ def __call__(self):
+    _sum = t1 + t2
+    _sum = np.where(_sum < 0.0, 0.0, _sum)
+
+    return 2.0*np.arcsin(np.sqrt(_sum))
+
+
+class Blob_survey(Greedy_survey_fields):
+    """Select observations in large, mostly contiuguous, blobs.
+    """
+    def __init__(self, basis_functions, basis_weights,
+                 extra_features=None, filtername='r', filter2='g',
+                 slew_approx=7.5, filter_change_approx=140.,
+                 read_approx=2., exptime=30., nexp=2,
+                 ideal_pair_time=22., min_pair_time=15.,
+                 search_radius =30., alt_max = 85., az_range=90.,
+                 smoothing_kernel=None, nside=default_nside,
+                 dither=True, seed=42, ignore_obs='ack',
+                 tag_fields=False, tag_map=None, tag_names=None,
+                 sun_alt_limit=-19., survey_note='blob',
+                 sitename='LSST'):
+        """
+        Parameters
+        ----------
+        filtername : str ('r')
+            The filter to observe in.
+        filter2 : str ('g')
+            The filter to pair with the first observation. If set to None, no pair
+            will be observed.
+        slew_approx : float (7.5)
+            The approximate slewtime between neerby fields (seconds). Used to calculate
+            how many observations can be taken in the desired time block.
+        filter_change_approx : float (140.)
+             The approximate time it takes to change filters (seconds).
+        ideal_pair_time : float (22.)
+            The ideal time gap wanted between observations to the same pointing (minutes)
+        min_pair_time : float (15.)
+            The minimum acceptable pair time (minutes)
+        search_radius : float (30.)
+            The radius around the reward peak to look for additional potential pointings (degrees)
+        alt_max : float (85.)
+            The maximum altitude to include (degrees).
+        az_range : float (90.)
+            The range of azimuths to consider around the peak reward value (degrees).
+        sitename : str ('LSST')
+            The name of the site to lookup latitude and longitude.
+        """
+
+        if nside is None:
+            nside = set_default_nside()
+
+        if extra_features is None:
+            extra_features = {}
+            extra_features['night'] = features.Current_night()
+            extra_features['mounted_filters'] = features.Mounted_filters()
+            extra_features['mjd'] = features.Current_mjd()
+            extra_features['night_boundaries'] = features.CurrentNightBoundaries()
+            extra_features['sun_moon_alt'] = features.Sun_moon_alts()
+            extra_features['lmst'] = features.Current_lmst()  # Pretty sure in hours
+            extra_features['current_filter'] = features.Current_filter()
+            extra_features['altaz'] = features.AltAzFeature()
+
+        super(Blob_survey, self).__init__(basis_functions=basis_functions,
```

Comment:
Need to pass `tag_fields=tag_fields, tag_map=tag_map, tag_names=tag_names,` to super so the functionality is preserved.